web/app/view/CustomNumberField.js

```diff
@@ -20,30 +20,79 @@ Ext.define('Traccar.view.CustomNumberField', {
     xtype: 'customNumberField',
 
     beforeEl: '<div style="width:100%;display:inline-table;">',
-    unitEl: '<div style="display:table-cell;padding-left:10px;vertical-align:middle;">',
+    unitEl: '<div id="numberUnitEl" style="display:table-cell;padding-left:10px;vertical-align:middle;width:1px;white-space:nowrap;">',
 
     constructor: function (config) {
-        var unit;
-        if (config.dataType === 'speed') {
-            unit = Traccar.app.getPreference('speedUnit', 'kn');
-            config.beforeSubTpl = this.beforeEl;
-            config.afterSubTpl = this.unitEl + Ext.getStore('SpeedUnits').findRecord('key', unit).get('name') + '</div></div>';
+        var unitName = '';
+        if (config.dataType) {
+            config.beforeBodyEl = this.beforeEl;
+            switch (config.dataType) {
+                case 'speed':
+                    config.units = {};
+                    config.units.getStore = function () {
+                        return Ext.getStore('SpeedUnits');
+                    };
+                    config.units.getValue = function () {
+                        return Traccar.app.getPreference('speedUnit', 'kn');
+                    };
+                    unitName = Ext.getStore('SpeedUnits').findRecord('key', config.units.getValue()).get('name');
+                    break;
+                case 'distance':
+                    config.units = {};
+                    config.units.getStore = function () {
+                        return Ext.getStore('DistanceUnits');
+                    };
+                    config.units.getValue = function () {
+                        return Traccar.app.getPreference('distanceUnit', 'km');
+                    };
+                    unitName = Ext.getStore('DistanceUnits').findRecord('key', config.units.getValue()).get('name');
+                    break;
+                case 'frequency':
```

Comment:
This will look like this:
![image](https://user-images.githubusercontent.com/5688080/30046866-3485aa80-9227-11e7-98fd-f2ae49a5d96c.png)
![image](https://user-images.githubusercontent.com/5688080/30046867-35b9f4ba-9227-11e7-89fa-854c6f891b62.png)